SonataCoreBundle.php

```diff
@@ -27,7 +28,12 @@ public function build(ContainerBuilder $container)
     {
         $container->addCompilerPass(new StatusRendererCompilerPass());
         $container->addCompilerPass(new AdapterCompilerPass());
-        $container->addCompilerPass(new FormFactoryCompilerPass());
+        if (class_exists(FormPass::class)) {
+            $container->addCompilerPass(new Compiler\FormFactoryCompilerPass());
+        }
+        else {
+            $container->addCompilerPass(new Compiler\LegacyFormFactoryCompilerPass());
+        }
```

Comment:
This will make StyleCi angry, please use PSR-2. Also, you could use a ternary inside `addCompilerPass` instead.